core/common/src/main/java/alluxio/conf/RemovedKey.java

```diff
@@ -0,0 +1,218 @@
+          .setScope(Scope.CLIENT)
+          .build();
+
+  @Removed
+  public static final PropertyKey USER_UFS_DELEGATION_READ_BUFFER_SIZE_BYTES =
+      new PropertyKey.Builder(PropertyKey.Name.USER_UFS_DELEGATION_READ_BUFFER_SIZE_BYTES)
+          .setDefaultValue("8MB")
+          .setDescription("Size of the read buffer when reading from the UFS through the "
+              + "Alluxio worker. Each read request will fetch at least this many bytes, "
+              + "unless the read reaches the end of the file.")
+          .setScope(Scope.CLIENT)
+          .build();
+
+  @Removed
+  public static final PropertyKey USER_UFS_DELEGATION_WRITE_BUFFER_SIZE_BYTES =
+      new PropertyKey.Builder(PropertyKey.Name.USER_UFS_DELEGATION_WRITE_BUFFER_SIZE_BYTES)
+          .setDefaultValue("2MB")
+          .setDescription("Size of the write buffer when writing to the UFS through the "
+              + "Alluxio worker. Each write request will write at least this many bytes, "
+              + "unless the write is at the end of the file.")
+          .setScope(Scope.CLIENT)
+          .build();
+
+  @Removed(message = "Use alluxio.worker.tieredstore.levelX.watermark.{high/low}.ratio instead")
+  public static final PropertyKey WORKER_TIERED_STORE_LEVEL0_RESERVED_RATIO =
+      new PropertyKey.Builder(PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_RESERVED_RATIO, 0)
+          .setDescription("Fraction of space reserved in the top storage tier. "
+              + "This has been deprecated, please use high and low watermark instead.")
+          .setScope(Scope.WORKER)
+          .build();
+
+  @Removed(message = "Use alluxio.worker.tieredstore.levelX.watermark.{high/low}.ratio instead")
+  public static final PropertyKey WORKER_TIERED_STORE_LEVEL1_RESERVED_RATIO =
+      new PropertyKey.Builder(PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_RESERVED_RATIO, 1)
+          .setDescription("Fraction of space reserved in the second storage tier. "
+              + "This has been deprecated, please use high and low watermark instead.")
+          .setScope(Scope.WORKER)
+          .build();
+
+  @Removed(message = "Use alluxio.worker.tieredstore.levelX.watermark.{high/low}.ratio instead")
+  public static final PropertyKey WORKER_TIERED_STORE_LEVEL2_RESERVED_RATIO =
+      new PropertyKey.Builder(PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_RESERVED_RATIO, 2)
+          .setDescription("Fraction of space reserved in the third storage tier. "
+              + "This has been deprecated, please use high and low watermark instead.")
+          .setScope(Scope.WORKER)
+          .build();
+
+  // Removed keys - never use these. They will throw an exception during validation
+  @Removed(message = "This key is used only for testing. It is always removed")
+  public static final PropertyKey TEST_REMOVED_KEY =
+      new PropertyKey.Builder("alluxio.test.removed.key")
+          .build();
+
+  private static final class Name {
+  }
```

Comment:
It can probably be removed. I was previously considering moving deprecated property key names into this class. What do you think about moving the removed names into here?

Reply:
This has been updated to include key names now